OrbitGl/Batcher.cpp

```diff
@@ -9,7 +9,7 @@
 
 void Batcher::AddLine(const Line& line, const Color* colors,
                       PickingID::Type picking_type,
-                      PickingUserData user_data) {
+                      std::shared_ptr<PickingUserData> user_data) {
```

Comment:
I would say someone needs to assume ownership over PickingUserData here. Maybe have PickingUserData passed as r-value so you can std::move() it to line_buffer.m_UserData, in which case the onwership will belong to m_UserData

Reply:
looking at the rest of the code it seems to be the case that once this is passed to AddLine function it is no longer user by the caller. I think you can do PickingUserData&& user_data here and use std::move() when passing it to this function.

Reply:
I actually wanted to use a unique_ptr here, but this would have required changes to the BlockChain (it always copies on "add"), and I did not want to introduce to many changes since it's quite fundamental. So I stuck with the shared_ptr, also because there may be situations where not taking ownership may actually arise... right now it's not used, correct.
Isn't it a bit counter-intuitive to use a shared_ptr, but then forcing to move it?

Reply:
well, the idea here is to not use shared_ptr - we do not really need reference counting here do we?

Reply:
Changed to use std::unique_ptr<>